app/uk/gov/hmrc/transitmovements/services/DeparturesService.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovements.services
+
+import cats.data.EitherT
+import com.google.inject.ImplementedBy
+import uk.gov.hmrc.transitmovements.models.responses.DeclarationResponse
+import uk.gov.hmrc.transitmovements.models.values.BytesToHex
+import uk.gov.hmrc.transitmovements.models.values.ShortUUID
+import uk.gov.hmrc.transitmovements.models.DeclarationData
+import uk.gov.hmrc.transitmovements.models.Departure
+import uk.gov.hmrc.transitmovements.models.DepartureId
+import uk.gov.hmrc.transitmovements.models.EORINumber
+import uk.gov.hmrc.transitmovements.repositories.DeparturesRepository
+
+import java.security.SecureRandom
+import java.time.Clock
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import javax.inject.Inject
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[DeparturesServiceImpl])
+trait DeparturesService {
+  def insertDeparture(declarationData: DeclarationData): EitherT[Future, Throwable, DeclarationResponse]
+}
+
+class DeparturesServiceImpl @Inject() (
+  repository: DeparturesRepository,
+  clock: Clock,
+  random: SecureRandom
+) extends DeparturesService {
+
+  def insertDeparture(declarationData: DeclarationData): EitherT[Future, Throwable, DeclarationResponse] =
+    repository.insert(createDeparture(declarationData))
+
+  val createDeparture: DeclarationData => Departure = {
```

Comment:
Is there a reason this isn't just a def?

Also, you'll need to accept the enrollment EORI from the controller, which should be receiving it via a URL parameter..

Reply:
Changed to a def, adding an EORINumber